src/backend/d3d12/PipelineD3D12.cpp

```diff
@@ -116,10 +118,14 @@ namespace d3d12 {
                 }
             }
 
-            InputState* inputState = ToBackend(GetInputState());
-            descriptor.InputLayout = inputState->GetD3D12InputLayoutDescriptor();
+            PipelineLayout* layout = ToBackend(GetLayout());
+
+            descriptor.pRootSignature = layout->GetRootSignature().Get();
 
-            descriptor.pRootSignature = ToBackend(GetLayout())->GetRootSignature().Get();
+            InputState* inputState = ToBackend(GetInputState());
+            if (inputState->GetAttributesSetMask().any()) {
```

Comment:
It's empty, and D3D prints a warning. It says that it's fine but you probably didn't intend to do it.